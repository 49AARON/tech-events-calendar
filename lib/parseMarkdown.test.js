const parseMarkdown = require('./parseMarkdown')
const s = require('common-tags').stripIndent

it('parses a valid markdown file successfully', () => {
  const md = s`
    ---
    id: google-cloud-onboard-bangkok-2018
    date: '2018-01-30'
    time:
      - '08:30 ~ 12:00'
      - '12:00 ~ 13:00 (break lol)'
      - '13:00 ~ 17:30++'
    location:
      title: Centara Grand at CentralWorld
      url: >-
        https://www.google.com/maps/place/Centara+Grand+at+CentralWorld/@13.7477222,100.5365583,17z/data=!3m1!4b1!4m5!3m4!1s0x30e2992f7809567f:0xccc050cff0e7d234!8m2!3d13.747717!4d100.538747
      detail: 'Convention Centre B2, 22nd Floor'
    categories:
      - Workshop
    topics:
      - Google Cloud Platform
      - Cloud
    links:
      - type: website
        url: 'https://cloudplatformonline.com/2018-onboard-bkk.html'
        title: Google Cloud
      - type: ticket
        url: 'https://cloudplatformonline.com/2018-onboard-bkk.html#register'
        title: Google Cloud

    ---
    # Google Cloud OnBoard Bangkok 2018

    > งานสัมมนา Google Cloud OnBoard นี้เป็นงานสัมมนาฟรีที่วิทยากรจาก Google
    > จะมาแนะนำเทคโนโลยีของ Google Cloud Platform งานนี้เหมาะกับทั้ง IT Managers,
    > Systems Engineers and Operations professionals, Developers, Solution
    > Architects, และผู้ที่สนใจในเทคโนโลยีของ Google ทุกคน

    Cloud OnBoard is a free full-day instructor-led enablement and training event that will provide you with a step-by-step technical introduction to the Google Cloud Platform (GCP). Through a combination of instructor-led presentations and hands-on labs, you will learn how to get started with Google App Engine, Datastore, Storage, Container Engine, Compute Engine and Network, Big Data and Machine Learning.
  `
  const result = parseMarkdown(md)
  expect(result).toMatchSnapshot()
})
