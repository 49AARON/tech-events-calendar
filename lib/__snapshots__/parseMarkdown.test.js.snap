// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses a valid markdown file successfully 1`] = `
Object {
  "checks": Array [
    Object {
      "children": Array [
        Object {
          "result": Object {
            "status": true,
          },
          "title": "is present",
          "type": "check",
        },
        Object {
          "result": Object {
            "status": true,
          },
          "title": "is a string",
          "type": "check",
        },
      ],
      "title": "\`id\`",
      "type": "item",
    },
    Object {
      "children": Array [
        Object {
          "result": Object {
            "status": true,
          },
          "title": "is present",
          "type": "check",
        },
        Object {
          "result": Object {
            "status": true,
          },
          "title": "is a string",
          "type": "check",
        },
        Object {
          "result": Object {
            "status": true,
          },
          "title": "is in form of \\"YYYY-MM-DD\\" or \\"YYYY-MM-DD ~ YYYY-MM-DD\\"",
          "type": "check",
        },
        Object {
          "result": Object {
            "status": true,
          },
          "title": "start date is valid",
          "type": "check",
        },
        Object {
          "result": Object {
            "status": true,
          },
          "title": "end date is valid",
          "type": "check",
        },
      ],
      "title": "\`date\`",
      "type": "item",
    },
    Object {
      "children": Array [
        Object {
          "result": Object {
            "status": true,
          },
          "title": "is present",
          "type": "check",
        },
        Object {
          "result": Object {
            "status": true,
          },
          "title": "is an array",
          "type": "check",
        },
        Object {
          "result": Object {
            "status": true,
          },
          "title": "item 0 is valid",
          "type": "check",
        },
      ],
      "title": "\`categories\`",
      "type": "item",
    },
    Object {
      "children": Array [
        Object {
          "result": Object {
            "status": true,
          },
          "title": "is present",
          "type": "check",
        },
        Object {
          "result": Object {
            "status": true,
          },
          "title": "is an array",
          "type": "check",
        },
        Object {
          "result": Object {
            "status": true,
          },
          "title": "item 0 is valid",
          "type": "check",
        },
        Object {
          "result": Object {
            "status": true,
          },
          "title": "item 1 is valid",
          "type": "check",
        },
      ],
      "title": "\`topics\`",
      "type": "item",
    },
    Object {
      "children": Array [
        Object {
          "result": Object {
            "status": true,
          },
          "title": "is an array of time strings",
          "type": "check",
        },
        Object {
          "children": Array [
            Object {
              "result": Object {
                "status": true,
              },
              "title": "is a string",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "is in form \\"HH:MM ~ HH:MM[++][ (agenda)]\\"",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "start time is valid",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "end time is valid",
              "type": "check",
            },
          ],
          "title": "item 0",
          "type": "item",
        },
        Object {
          "children": Array [
            Object {
              "result": Object {
                "status": true,
              },
              "title": "is a string",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "is in form \\"HH:MM ~ HH:MM[++][ (agenda)]\\"",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "start time is valid",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "end time is valid",
              "type": "check",
            },
          ],
          "title": "item 1",
          "type": "item",
        },
        Object {
          "children": Array [
            Object {
              "result": Object {
                "status": true,
              },
              "title": "is a string",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "is in form \\"HH:MM ~ HH:MM[++][ (agenda)]\\"",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "start time is valid",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "end time is valid",
              "type": "check",
            },
          ],
          "title": "item 2",
          "type": "item",
        },
      ],
      "title": "time",
      "type": "item",
    },
    Object {
      "children": Array [
        Object {
          "result": Object {
            "status": true,
          },
          "title": "markdown body is valid",
          "type": "check",
        },
        Object {
          "result": Object {
            "status": true,
          },
          "title": "title is present",
          "type": "check",
        },
      ],
      "title": "title",
      "type": "item",
    },
    Object {
      "children": Array [
        Object {
          "result": Object {
            "status": true,
          },
          "title": "is present",
          "type": "check",
        },
        Object {
          "result": Object {
            "status": true,
          },
          "title": "\`title\` is a string",
          "type": "check",
        },
        Object {
          "result": Object {
            "status": true,
          },
          "title": "\`url\` is an optional Google Maps URL",
          "type": "check",
        },
        Object {
          "result": Object {
            "status": true,
          },
          "title": "\`detail\` is an optional string",
          "type": "check",
        },
      ],
      "title": "\`location\`",
      "type": "item",
    },
    Object {
      "children": Array [
        Object {
          "result": Object {
            "status": true,
          },
          "title": "summary",
          "type": "check",
        },
      ],
      "title": "summary",
      "type": "item",
    },
    Object {
      "children": Array [
        Object {
          "result": Object {
            "status": true,
          },
          "title": "description",
          "type": "check",
        },
      ],
      "title": "description",
      "type": "item",
    },
    Object {
      "children": Array [
        Object {
          "result": Object {
            "status": true,
          },
          "title": "is an array",
          "type": "check",
        },
        Object {
          "children": Array [
            Object {
              "result": Object {
                "status": true,
              },
              "title": "\`type\` is valid",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "\`url\` is a string",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "\`url\` is an http(s) url",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "\`detail\` is an optional string",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "\`price\` is an optional string (only present in ticket)",
              "type": "check",
            },
          ],
          "title": "item 0",
          "type": "item",
        },
        Object {
          "children": Array [
            Object {
              "result": Object {
                "status": true,
              },
              "title": "\`type\` is valid",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "\`url\` is a string",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "\`url\` is an http(s) url",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "\`detail\` is an optional string",
              "type": "check",
            },
            Object {
              "result": Object {
                "status": true,
              },
              "title": "\`price\` is an optional string (only present in ticket)",
              "type": "check",
            },
          ],
          "title": "item 1",
          "type": "item",
        },
      ],
      "title": "links",
      "type": "item",
    },
  ],
  "event": Object {
    "categories": Array [
      "Workshop",
    ],
    "end": Object {
      "date": 30,
      "month": 1,
      "year": 2018,
    },
    "id": "google-cloud-onboard-bangkok-2018",
    "links": Array [
      Object {
        "title": "Google Cloud",
        "type": "website",
        "url": "https://cloudplatformonline.com/2018-onboard-bkk.html",
      },
      Object {
        "title": "Google Cloud",
        "type": "ticket",
        "url": "https://cloudplatformonline.com/2018-onboard-bkk.html#register",
      },
    ],
    "location": Object {
      "detail": "Convention Centre B2, 22nd Floor",
      "title": "Centara Grand at CentralWorld",
      "url": "https://www.google.com/maps/place/Centara+Grand+at+CentralWorld/@13.7477222,100.5365583,17z/data=!3m1!4b1!4m5!3m4!1s0x30e2992f7809567f:0xccc050cff0e7d234!8m2!3d13.747717!4d100.538747",
    },
    "start": Object {
      "date": 30,
      "month": 1,
      "year": 2018,
    },
    "summary": "Cloud OnBoard is a free full-day instructor-led enablement and training event that will provide you with a step-by-step technical introduction to the Google Cloud Platform (GCP). Through a combination of instructor-led presentations and hands-on labs, you will learn how to get started with Google App Engine, Datastore, Storage, Container Engine, Compute Engine and Network, Big Data and Machine Learning.",
    "time": Array [
      Object {
        "after": false,
        "agenda": null,
        "from": Object {
          "hour": 8,
          "minute": 30,
        },
        "to": Object {
          "hour": 12,
          "minute": 0,
        },
      },
      Object {
        "after": false,
        "agenda": "break lol",
        "from": Object {
          "hour": 12,
          "minute": 0,
        },
        "to": Object {
          "hour": 13,
          "minute": 0,
        },
      },
      Object {
        "after": true,
        "agenda": null,
        "from": Object {
          "hour": 13,
          "minute": 0,
        },
        "to": Object {
          "hour": 17,
          "minute": 30,
        },
      },
    ],
    "title": "Google Cloud OnBoard Bangkok 2018",
    "topics": Array [
      "Google Cloud Platform",
      "Cloud",
    ],
  },
}
`;
